pkg/test/useraccount/assertion.go

```diff
@@ -51,10 +51,21 @@ func (a *Assertion) Exists() *Assertion {
 	return a
 }
 
+func (a *Assertion) HasEmbeddedSpec(spec toolchainv1alpha1.UserAccountSpecEmbedded, disabled bool, userID string) *Assertion {
+	err := a.loadUaAssertion()
+	require.NoError(a.t, err)
+	assert.EqualValues(a.t, spec.UserAccountSpecBase, a.userAccount.Spec.UserAccountSpecBase)
```

Comment:
I believe that `assert.Equal(a.t, spec, a.userAccount.Spec)` will do the work (I uses `reflect.DeepEqual(a, b)` under the hood, so you don't need to specify all fields one by one